test/apps/iodemo/io_demo.cc

```diff
@@ -172,21 +250,75 @@ class IoDemoRandom {
         }
     }
 
-    static inline void fill(unsigned &seed, void *buffer, size_t size) {
+    static void *get_host_fill_buffer(void *buffer, size_t size,
+                                      ucs_memory_type_t memory_type)
+    {
+#ifdef HAVE_CUDA
+        void *buffer_p = buffer;
+        static std::vector<uint8_t> _buffer;
+
+        if (memory_type == UCS_MEMORY_TYPE_CUDA) {
+            _buffer.resize(size);
+            buffer_p = _buffer.data();
+        }
+
+        return buffer_p;
+#else
+        return buffer;
+#endif
+    }
+
+    static void fill_commit(void *buffer, void *fill_buffer, size_t size,
+                            ucs_memory_type_t memory_type)
+    {
+#ifdef HAVE_CUDA
+        if (memory_type == UCS_MEMORY_TYPE_CUDA) {
+            cudaMemcpy(buffer, fill_buffer, size, cudaMemcpyDefault);
+        }
+#endif
+    }
```

Comment:
can we avoid this wrapper?

Reply:
We can, but do we want?
My rationale was to put the "ugliness" of `#ifdef`s in one single point, separated from the function that has the logic (`fill()`) so that it looks cleaner.

Reply:
ok